packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,140 @@
+      if (this.sourceFsPath) {
+        if (this.responseText) {
+          renameComponent(this.sourceFsPath, this.responseText);
+          return true;
+        }
+      }
+      return false;
+  }
+}
+
+export async function forceRenameLightningComponent(sourceUri: vscode.Uri) {
+  const sourceFsPath = sourceUri.fsPath;
+  const inputOptions = {
+    placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+    promopt: nls.localize(RENAME_INPUT_PROMPT)
+  } as vscode.InputBoxOptions;
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = path.basename(componentPath);
+  checkForDuplicateName(componentPath, newName);
+  const items = fs.readdirSync(componentPath);
+  for (const item of items) {
+    // only rename the file that has same name with component
+    if (isNameMatch(item, componentName, componentPath)) {
+      const newItem = item.replace(componentName, newName);
+      fs.renameSync(
+        path.join(componentPath, item),
+        path.join(componentPath, newItem)
+      );
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  fs.renameSync(
+    componentPath,
+    newComponentPath
+  );
+  notificationService.showWarningMessage(nls.localize(RENAME_WARNING));
+}
+
+function getComponentPath(sourceFsPath: string): string {
+  const stats = fs.statSync(sourceFsPath);
+  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+}
+
+function checkForDuplicateName(componentPath: string, newName: string) {
+  if (isDuplicate(componentPath, newName)) {
+    const errorMessage = nls.localize(RENAME_INPUT_DUP_ERROR);
+    notificationService.showErrorMessage(errorMessage);
+    throw new Error(format(errorMessage));
+  }
+}
+
+function isDuplicate(componentPath: string, newName: string): boolean {
+  // A LWC component can't share the same name as a Aura component
+  const componentPathDirName = path.dirname(componentPath);
+  let lwcPath: string;
+  let auraPath: string;
+  if (isLwcComponent(componentPath)) {
+    lwcPath = componentPathDirName;
+    auraPath = path.join(path.dirname(componentPathDirName), 'aura');
+  } else {
+    lwcPath = path.join(path.dirname(componentPathDirName), 'lwc');
+    auraPath = componentPathDirName;
+  }
+  const allLwcComponents = fs.readdirSync(lwcPath);
+  const allAuraComponents = fs.readdirSync(auraPath);
+  if (allLwcComponents.includes(newName) || allAuraComponents.includes(newName)) {
+    return true;
+  }
+  return false;
+}
+
+function isNameMatch(item: string, componentName: string, componentPath: string) {
```

Comment:
we should have a test for this method.

Reply:
@gbockus-sf Updated the test file to include this function.